src/main/java/org/terasology/launcher/config/ConfigValidator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.util.JavaHeapSize;
+
+/**
+ * Provides validated configurations by resetting to
+ * default values in case any configuration is found
+ * invalid.
+ */
+class ConfigValidator {
+    private static final Logger logger = LoggerFactory.getLogger(ConfigValidator.class);
+
+    private final Config defaultConfig;
```

Comment:
Maybe call this `fallbackConfig`, but that's definitely a matter of taste I would say.